api/v1beta1/verticadb_types.go

```diff
@@ -1253,3 +1258,9 @@ func (v *VerticaDB) IsHTTPServerEnabled() bool {
 func (v *VerticaDB) IsEON() bool {
 	return v.Spec.ShardCount > 0
 }
+
+// IsAgentEnabled returns true if the annotation to enable the argent
+// has been set to the correct value
+func (v *VerticaDB) IsAgentEnabled() bool {
+	return strings.EqualFold(v.Spec.Annotations[StartAgent], AgentEnabled)
```

Comment:
Lets check the annotations that are present in the vdb's metadata. The annotations that are set in the `spec` apply to objects that are created by the operator.
```suggestion
	return strings.EqualFold(v.ObjectMeta.Annotations[StartAgent], AgentEnabled)
```